triton-client/src/components/DatasetCard.tsx

```diff
@@ -46,6 +54,15 @@ function DatasetCard({ datasetID }: DatasetCardProps) {
 		dispatch(fetchReadsets(datasetID))
 	}, [datasetID, dispatch])
 
+  useEffect(() => {
+    if (dataset && dataset.requests.length > 0) {
+      setActiveRequest(dataset?.requests[0])
+    }
+    else {
+      setActiveRequest(undefined)
+    }
+  },[dataset, dataset?.requests])
+
```

Comment:
Doing this in a `useEffect` is unnecessary.
You can set `activeRequest` to
```ts
const activeRequest = dataset && dataset.requests.length > 0 ? dataset.requests[0] : undefined
```

Reply:
The whole component will re-render anyway if dataset changes.

Reply:
Another option could be to use `useMemo`, but for that expression it might not be worthwhile.

Reply:
yeah but requests may change without dataset changing...

Reply:
In redux, you're actually supposed to create a new data each time, but we kinda avoid that explicitly using immer.

Reply:
I believe changes are detected recursively?
But anyway, `useMemo` can still help with that; you just pass `data.requests` as a dependency.

Reply:
eheh i will use a useMemo ...